### ISSUE TITLE
[Feature]: Request for a Connection Status Indicator for --share Users.

### ISSUE THREAD

**Issue by Nyx01** [vladmandic/sdnext#3164]
### Feature description

I am reposting this here from discord as requested;

A feature request for a status indicator to show connection/connection stats for users in --share mode.
Dropping off with spotty wifi or bouncing between a 4g/5g network can sometimes cause the client browser to become unresponsive (or crash occasionally!), but if you can catch it early and refresh, usually it'll recover just fine. A status indicator would be amazing to have to let you know if you've dropped or need to refresh to reconnect properly.  (Also perhaps a better method of reconnecting could be considered, as well as better updating progress indicator on generations that doesn't rely on the visual preview for slower remote connections?...)

**Comment by Aptronymist** [comment]
I agree this would be useful info to have considering how many times I've accessed people's shares to fix their issues and whatnot.

**Comment by vladmandic** [comment]
i agree with the request very much, i just don't know how to do the check.
what fails is websocket created inside gradio itself that i have no access to. if i did, i could catch any possible error states or check status. but its internal variable that i have no access to. i'll leave the request open, maybe i figure something out.

**Comment by vladmandic** [comment]
took forever, but finally got to this:

<img width="444" height="119" alt="Image" src="https://github.com/user-attachments/assets/8748a222-4202-4dec-8302-38df81aa8343" />
